recipes/cgal/all/conanfile.py

```diff
@@ -42,40 +31,28 @@ def _build_subfolder(self):
     def _configure_cmake(self):
         if not self._cmake:
             self._cmake = CMake(self)
-            self._cmake.definitions["WITH_CGAL_Core"] = self.options.with_cgal_core
-            self._cmake.definitions["WITH_CGAL_Qt5"] = self.options.with_cgal_qt5
-            self._cmake.definitions["WITH_CGAL_ImageIO"] = self.options.with_cgal_imageio
-            self._cmake.definitions["CGAL_HEADER_ONLY"] = self.options.header_only
+            self._cmake.definitions["CGAL_HEADER_ONLY"] = "TRUE"
             self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def _patch_sources(self):
-        if tools.Version(self.version) < "5.3":
-            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+        if scm.Version(self.version) < "5.3":
+            files.replace_in_file(self, os.path.join(self._source_subfolder, "CMakeLists.txt"),
                                 "CMAKE_SOURCE_DIR", "CMAKE_CURRENT_SOURCE_DIR")
         else:
-            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+            files.replace_in_file(self, os.path.join(self._source_subfolder, "CMakeLists.txt"),
                                 "if(NOT PROJECT_NAME)", "if(TRUE)")
 
-    def configure(self):
-        if self.options.with_cgal_qt5:
-            raise ConanInvalidConfiguration("Qt Conan package is not available yet.")
-        if self.options.header_only:
-            del self.options.shared
-
     def requirements(self):
         self.requires("boost/1.75.0")
         self.requires("eigen/3.3.9")
         self.requires("mpfr/4.1.0")
 
     def package_id(self):
-        if self.options.header_only:
-            self.info.header_only()
+        self.info.header_only()
```

Comment:
```suggestion
        self.info.clear()
```
header_only will be deprecated soon: https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-package-id-method